mapproxy/script/conf/app.py

```diff
@@ -71,7 +71,7 @@ def write_header(f, capabilities):
 @contextmanager
 def file_or_stdout(name):
     if name == '-':
-        yield codecs.getwriter('utf-8')(sys.stdout)
+        yield codecs.getwriter('utf-8')(sys.stdout.buffer)
```

Comment:
Caution, the integration tests failed on your modifications.
I think your line of code should be:
```
yield codecs.getwriter('utf-8')(sys.stdout.decode())
```

background: in python2 there was little distinction between byte streams and string. unicode was treated separately.
In python3, you go from bytes to str (unicode) via decode() and from str to bytes via encode()
https://stackoverflow.com/questions/14472650/python-3-encode-decode-vs-bytes-str

Reply:
That doesn't make any sense. We're writing to stdout, not reading from it.

```python
>>> sys.stdout.decode
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: '_io.TextIOWrapper' object has no attribute 'decode'
```

Reply:
yes, you're completely right, I didn't read the error message thoroughly enough. So maybe there is something wrong with the CI test, the fixtures, should have a look into that.
Probably your fix is correct, since you're testing it on a real life example. However something does not work in the automatic tests.

Reply:
There seems to be some trouble with using stdout in CI tests, for example a special case is made here:
https://github.com/mapproxy/mapproxy/blob/89217d05a2903c98a0e4bf7f91ac5e2dab5d94b0/mapproxy/script/conf/app.py#L48

I think your solution works fine on python3, but in python2 sys.stdout does not have a buffer attribute. It would be better to find a solution that works for both python2 and 3.

Reply:
https://github.com/mapproxy/mapproxy/pull/521/commits/99a8a1cb847112863cc685757057d800fd3d6d28 should fix this problem both in the tests and in Python2

Reply:
great! looks like a good hack.
Strange that CI has not started on your push.

Reply:
Take a look at https://travis-ci.org/github/mapproxy/mapproxy/requests - it says "Since June 15th, 2021, the building on travis-ci.org is ceased. Please use travis-ci.com from now on.".  Travis got the request but it rejected the job as Travis is in "Read-only mode"